src/presentation/reader.rs

```diff
@@ -185,12 +180,12 @@ impl SessionManager {
     /// Internally it generates the ephemeral keys,
     /// derives the shared secret, and derives the session keys
     /// (using **Diffie–Hellman key exchange**).
-    pub fn establish_session(
+    pub async fn establish_session(
         qr_code: String,
         namespaces: device_request::Namespaces,
         trust_anchor_registry: Option<TrustAnchorRegistry>,
         reader_x5chain: X5Chain,
-        reader_key: &str,
+        reader_signer: impl FnOnce(Vec<u8>) -> Pin<Box<dyn Future<Output = Result<Vec<u8>>> + Send>>,
```

Comment:
NOTE: Ephemeral keys are created within the session manager, and there is no need for an external signer for the DH key exchange, outside of the session manager's control.